File: RelacjeSportowe.DataAccess/Dtos/Requests/RegisterUserRequest.cs
using System.ComponentModel.DataAnnotations;

namespace RelacjeSportowe.DataAccess.Dtos.Requests
{
    public class RegisterUserRequest
    {
        [Required]
        public string Email { get; set; }

        [Required]
        public string Username { get; set; }

        [Required]
        public string Password { get; set; }
    }
}

File: RelacjeSportowe.Web/ClientApp/src/app/models/dtos/requests/register-user-request.ts
export class RegisterUserRequest {
  public email: string;
  public username: string;
  public password: string;

  constructor(registerUserRequest: any) {
    this.email = registerUserRequest.email;
    this.username = registerUserRequest.username;
    this.password = registerUserRequest.password;
  }
}